Content.Server/_Moffstation/GameTicking/Rules/GunGameRuleSystem.cs

```diff
@@ -0,0 +1,272 @@
+    private void SpawnCurrentWeapon(NetUserId userId, GunGameRuleComponent rule)
+    {
+        if (!_mind.TryGetMind(userId, out var _, out var mind))
+            return;
+
+        if (mind.OwnedEntity is not { } playerEntity)
+            return;
+
+        var gear = EnsureComp<GunGameTrackerComponent>(playerEntity);
+        DeleteCurrentWeapons(gear);
+
+        if (!rule.PlayerRewards.TryGetValue(userId, out var gearQueue))
+        {
+            gearQueue = new Queue<EntityTableSelector>(rule.RewardSpawnsQueue);
+            rule.PlayerRewards.Add(userId, gearQueue);
+        }
+
+        if (gearQueue.Count == 0) // If we somehow try to spawn somebody's weapon after they win
+            return;
+
+        var slotTryOrder = new Queue<string>(rule.SlotTryOrder);
+        // Peek the player's queue, and spawn their items
+        foreach (var item in _entityTables.GetSpawns(gearQueue.Peek()))
+        {
+            SpawnItemAndEquip(
+                (playerEntity, gear),
+                item,
+                slotTryOrder);
+        }
+    }
+
+    /// <summary>
+    /// Attempts to spawn an item on a player in either their hands or the next available equipment slot according to
+    /// <see cref="Components.GunGameRuleComponent.SlotTryOrder"/>.
+    /// </summary>
+    private void SpawnItemAndEquip(Entity<GunGameTrackerComponent> player,
+        EntProtoId itemProto,
+        Queue<string> slotTryOrder)
+    {
+        var itemEnt = Spawn(itemProto);
+        player.Comp.CurrentRewards.Add(itemEnt);
+        while (slotTryOrder.TryDequeue(out var slot))
+        {
+            if (slot == "hand" && _hands.TryForcePickupAnyHand(player, itemEnt)
+                || _inventory.TryEquip(player, player, itemEnt, slot, true, true))
+                return;
+        }
+
+        // if there's no slots that work, drop it on the ground
+        _transform.SetCoordinates(itemEnt,
+            new EntityCoordinates(player,
+                _random.NextFloat() - 0.5f,
+                _random.NextFloat() - 0.5f));
+        _transform.SetLocalRotation(itemEnt, _random.NextAngle());
+    }
+
+    /// <summary>
+    /// Formats the scoreboard for the end of round screen.
+    /// </summary>
+    /// <returns>
+    /// The formatted message to add to the round-end screen.
+    /// Returns an empty message if the component doesn't exist.
+    /// </returns>
+    private FormattedMessage GetScoreboard(Entity<GunGameRuleComponent?> entity)
+    {
+        var msg = new FormattedMessage();
+
+        if (!Resolve(entity, ref entity.Comp))
+            return msg;
+
+        foreach (var (idx, (id, playerQueue)) in entity.Comp.PlayerRewards.OrderBy(p => p.Value.Count).Index())
+        {
+            if (!_player.TryGetPlayerData(id, out var data))
+                continue;
+
+            msg.AddMarkupOrThrow(Loc.GetString("gun-game-scoreboard-list-entry",
+                ("place", idx),
```

Comment:
I assume the LINQ indexing is index 0, so this'll make the first place `0th`, right?